File: javascript/8kyu/Sum of differences in array.js
// Your task is to sum the differences between consecutive pairs in the array in descending order.
// Example

// [2, 1, 10]  -->  9

// In descending order: [10, 2, 1]

// Sum: (10 - 2) + (2 - 1) = 8 + 1 = 9

// If the array is empty or the array has only one element the result should be 0 (Nothing in Haskell, None in Rust).

// https://www.codewars.com/kata/5b73fe9fb3d9776fbf00009e

function sumOfDifferences(arr) {
    arr.sort((a,b) => b - a);
    let sum = 0;
    for (let i = 0; i < arr.length - 1; ++i) {
        sum += arr[i] - arr[i + 1];
    }
    return sum;
}

File: java/8kyu/Sum of differences in array.java
// Your task is to sum the differences between consecutive pairs in the array in descending order. 

// https://www.codewars.com/kata/5b73fe9fb3d9776fbf00009e

import java.util.Arrays;

public class ZywOo {
  public static int sumOfDifferences(int[] arr) {
    Arrays.sort(arr);
		return arr.length <= 1 ? 0 : arr[arr.length-1] - arr[0];
  }
}